deployment/cre/jobs/operations/propose_gateway_job.go

```diff
@@ -0,0 +1,188 @@
+package operations
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/Masterminds/semver/v3"
+
+	chainsel "github.com/smartcontractkit/chain-selectors"
+
+	cldf "github.com/smartcontractkit/chainlink-deployments-framework/deployment"
+	"github.com/smartcontractkit/chainlink-deployments-framework/operations"
+	jobv1 "github.com/smartcontractkit/chainlink-protos/job-distributor/v1/job"
+	nodev1 "github.com/smartcontractkit/chainlink-protos/job-distributor/v1/node"
+	"github.com/smartcontractkit/chainlink-protos/job-distributor/v1/shared/ptypes"
+
+	"github.com/smartcontractkit/chainlink/deployment/cre/jobs/pkg"
+	"github.com/smartcontractkit/chainlink/deployment/cre/pkg/offchain"
+)
+
+type ChainSelector uint64
+
+func (cs *ChainSelector) UnmarshalText(data []byte) error {
+	ui, err := strconv.ParseUint(string(data), 10, 64)
+	if err != nil {
+		return err
+	}
+
+	*cs = ChainSelector(ui)
+	return nil
+}
+
+func (cs ChainSelector) MarshalText() ([]byte, error) {
+	return []byte(strconv.FormatUint(uint64(cs), 10)), nil
+}
+
+type ProposeGatewayJobInput struct {
+	Domain                  string
+	DONFilters              []offchain.TargetDONFilter
+	DONs                    []DON         `toml:"dons"`
+	GatewayKeyChainSelector ChainSelector `toml:"gateway_key_chain_selector"`
```

Comment:
why are we adding `toml` annotations here?

Reply:
Ah good catch! They should be YAML ones